src/internal/container-queries/use-resize-observer.ts

```diff
@@ -1,6 +1,7 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+import { unstable_batchedUpdates } from 'react-dom';
```

Comment:
Note that react-dom is a dev dependency —is this okay? Consumers of this package might not get it installed

Reply:
We already use it in this package: https://github.com/search?q=repo%3Acloudscape-design%2Fcomponent-toolkit%20unstable_batchedUpdates&type=code